charts/nginx-echo-headers/templates/ingress/ingressroutes.yaml

```
@@ -0,0 +1,52 @@
+{{- $httpsEnabled := .Values.ingress.https.enabled -}}
+{{- $fullname := include "nginx-echo-headers.fullname" . -}}
+{{- $servicePort := .Values.service.port -}}
+{{- $redirectMiddlewares := .Values.ingress.https.redirectMiddlewares -}}
+
+{{- if .Values.ingress.enabled }}
+{{- if $httpsEnabled }}
+---
+apiVersion: traefik.containo.us/v1alpha1
+kind: IngressRoute
+metadata:
+  name: {{ include "nginx-echo-headers.fullname" . }}
+spec:
+  entryPoints:
+  - websecure
+  routes:
+  {{- range .Values.ingress.dnsNames }}
+  - kind: Rule
+    match: Host(`{{.}}`)
+    priority: 10
+    services:
+    - name: {{ $fullname }}-https
+      port: {{ $servicePort }}
+  {{- end }}
+  tls:
+    secretName: {{ include "nginx-echo-headers.fullname" . }}
+{{- end }}
+---
+apiVersion: traefik.containo.us/v1alpha1
+kind: IngressRoute
+metadata:
+  name: {{ include "nginx-echo-headers.fullname" . }}-http
+spec:
+  entryPoints:
+    - web
+  routes:
+  {{- range .Values.ingress.dnsNames }}
+  - kind: Rule
+    match: Host(`{{.}}`)
+    priority: 10
+    services:
+    - name: {{ $fullname }}-https
+      port: {{ $servicePort }}
+    {{- if  $httpsEnabled }}
+    middlewares:
+    {{-  range $redirectMiddlewares }}
+    - name: {{ .name }}
+      namespace: {{ .namespace }}
```

Just for context, I've installed the redirect middleware in the Traefik namespace so you can consume that. 
```suggestion
      middlewares:
      - name: redirect-https
        namespace: kube-system
```

---

If you want, you could change the `$redirectMiddlewares` to `$middlewares` if you wanted to add more.

---

I don't think namespace should ever be coded into a chart. It should alway be picked up from where it installed

---

I like renaming it to `$middlewares` but I'm honestly not pro hardcoding this to our installed middleware!

---

@m-u this is the namespace where the middelware is installed. It's not guaranteed they'd be in the same namespace as the release in this case. I like to make it optional there but default to release namespace - my last commit does that

---

@m-u this is specific for the `redirect-https` middleware. https://github.com/untitledstartup/gitops-tools/blob/main/apps/traefik/tools-production/middlewares.yaml - it's already created here - agree to not hardcode the whole namespace and normally uses `Release.Namespace` traditionally, see [link](https://helm.sh/docs/chart_template_guide/builtin_objects/)